packages/shared/ConsolePatchingDev.js

```diff
@@ -20,13 +21,16 @@ let prevError;
 let prevGroup;
 let prevGroupCollapsed;
 let prevGroupEnd;
+const {
+  __REACT_DEVTOOLS_ENABLE_DOUBLE_LOGGING__: enableDoubleLogging,
```

Comment:
This still reads the value only once at startup. So switching it midway would not work. I think you want to update this value each time we enter `disableLogs` and `disabledDepth === 0`.

Reply:
Please, can i get your input on this. I tried but it seems  `getEnableDoubleLogging()` doesn't update

Reply:
@gaearon

Reply:
i tried this approach, 


```export function disableLogs(): void {
  if (__DEV__) {
    if (disabledDepth === 0) {
      /* eslint-disable react-internal/no-production-logging */
      console.log(another, '====>');
      try {
        if (!another) {
          patchedConsole();
        }
      } finally {
        another = suppressDoubleLogging;
      }
    }
    /* eslint-enable react-internal/no-production-logging */
  }
  disabledDepth++;
}```

Reply:
@gaearon 

```
function patchConsole() {
  prevLog = console.log;
  prevInfo = console.info;
  prevWarn = console.warn;
  prevError = console.error;
  prevGroup = console.group;
  prevGroupCollapsed = console.groupCollapsed;
  prevGroupEnd = console.groupEnd;
  // https://github.com/facebook/react/issues/19099
  const props = {
    configurable: true,
    enumerable: true,
    value: disabledLog,
    writable: true,
  };
  // $FlowFixMe Flow thinks console is immutable.
  Object.defineProperties(console, {
    info: props,
    log: props,
    warn: props,
    error: props,
    group: props,
    groupCollapsed: props,
    groupEnd: props,
  });
}

Reply:
```
export function getSuppressDoubleLogging(): boolean {
  try {
    const raw = localStorageGetItem(LOCAL_STORAGE_SHOULD_ENABLE_DOUBLE_LOGGING);
    if (raw != null) {
      return JSON.parse(raw);
    }
  } catch (error) {}
  return false;
}